api/golang/engine/lib/kurtosis_context/kurtosis_context.go

```diff
@@ -347,13 +349,18 @@ func newEnclaveContextFromEnclaveInfo(
 	portalClient portal_api.KurtosisPortalClientClient,
 	enclaveInfo *kurtosis_engine_rpc_api_bindings.EnclaveInfo,
 ) (*enclaves.EnclaveContext, error) {
-	// for remote contexts, we need to tunnel the APIC port to the local machine
-	if portalClient != nil {
-		apicGrpcPort := enclaveInfo.GetApiContainerHostMachineInfo().GetGrpcPortOnHostMachine()
-		forwardApicPortArgs := portal_constructors.NewForwardPortArgs(apicGrpcPort, apicGrpcPort, &apicPortTransportProtocol)
-		if _, err := portalClient.ForwardPort(ctx, forwardApicPortArgs); err != nil {
-			return nil, stacktrace.Propagate(err, "Unable to forward remote API container port to the local machine")
+	currentContext, err := store.GetContextsConfigStore().GetCurrentContext()
+	if err == nil {
+		// for remote contexts, we need to tunnel the APIC port to the local machine
+		if store.IsRemote(currentContext) && portalClient != nil {
+			apicGrpcPort := enclaveInfo.GetApiContainerHostMachineInfo().GetGrpcPortOnHostMachine()
+			forwardApicPortArgs := portal_constructors.NewForwardPortArgs(apicGrpcPort, apicGrpcPort, &apicPortTransportProtocol)
+			if _, err := portalClient.ForwardPort(ctx, forwardApicPortArgs); err != nil {
+				return nil, stacktrace.Propagate(err, "Unable to forward remote API container port to the local machine")
+			}
 		}
+	} else {
+		logrus.Warnf("Unable to retrieve current Kurtosis context. This is not critical, it will assume using Kurtosis default context for now.")
```

Comment:
I'd say we should throw here. Assuming default context if it's not can lead to unclear and confusing errors with the portal and stuffs. In the vein of failing early, we should prob throw